lib/pbench/test/unit/agent/test_config.py

```diff
@@ -34,3 +38,72 @@ def test_results_attr():
         "pbench"
         in PbenchAgentConfig(os.environ["_PBENCH_AGENT_CONFIG"]).results["user"]
     )
+
+
+# Monkeypatching different Classes and Methods to test Pbench-run value
+
+
+class MockPbenchAgentConfig:
+    """Mock class for PbenchAgentConfig"""
+
+    def __init__(self, cfg_name: str):
+        self.pbench_run = "/test/pbench-run"
+        self.pbench_log = "logger"
+        self.pbench_install_dir = Path("/test/install_dir")
+        self.ssh_opts = "/ssh_opts"
+        self.scp_opts = "/scp_opts"
+
+
+def mock_exist(path: Path):
+    """mock method for Path.exist"""
+    return True
```

Comment:
This is a slightly weak approach.  While, it's good enough for `test_pbench_run_path()`, in general, it would be better if it confirmed that `path` it receives matches the expected value (e.g., the value of `pbench_install_dir`).  (But, I think it's OK as is.)

Reply:
Yeah, I had similar thoughts. The scope of the test is limited, but but the methodology bothered me anyway and I ended up undecided about commenting. (Which is why I hadn't posted comments yesterday!)